### PR TITLE
Add architecture flags for MSVC

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes https://github.com/kokkos/kokkos/issues/5663. Note that I haven't tested this myself either.

### CODE REVIEW

**Review by crtrott — APPROVED** [review]

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -239,6 +239,7 @@ IF (KOKKOS_ARCH_ARMV80)
   COMPILER_SPECIFIC_FLAGS(
     COMPILER_ID KOKKOS_CXX_HOST_COMPILER_ID
     Cray    NO-VALUE-SPECIFIED
+    MSVC    NO-VALUE-SPECIFIED
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:242]:
```suggestion
    MSVC    /arch:armv8.0
```

File: cmake/kokkos_arch.cmake [hunk 2]
```diff
@@ -248,6 +249,7 @@ IF (KOKKOS_ARCH_ARMV81)
...
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:252]:
```suggestion
    MSVC    /arch:armv8.1
```

**Comment by dalg24** [comment]
Failures are unrelated (machine out of disk space)

**Review by dalg24 — APPROVED** [review]
Please make sure you update the changelog

**Pull request merged**